torch/csrc/jit/passes/onnx/eval_peephole.cpp

```diff
@@ -0,0 +1,155 @@
+#include <torch/csrc/jit/passes/onnx/eval_peephole.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <torch/torch.h>
+
+#include <c10/util/Optional.h>
+#include <algorithm>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+std::vector<at::Tensor> getValues(
+    Node* node,
+    const ValueToParamPairMap& valsToParamsMap) {
+  size_t numInputs = node->inputs().size();
+  std::vector<at::Tensor> inputTensorValues;
+  inputTensorValues.reserve(numInputs);
+  for (auto val : node->inputs()) {
+    if (val->node()->kind() == prim::Param) {
+      auto itr = valsToParamsMap.find(val);
+      if (itr == valsToParamsMap.end()) {
+        continue;
+      }
+      inputTensorValues.push_back(itr->second.second.toTensor());
+    } else if (val->node()->kind() == onnx::Constant) {
+      inputTensorValues.push_back(val->node()->t(attr::value));
+    } else {
+      continue;
+    }
+  }
+  return inputTensorValues;
+}
+
+// This pass fuses Conv and BatchNorm into Conv node
+// Conv and BatchNorm can be fused only if inputs for Batchnorm node:
+// scale, bias, mean and var are all tensors of same shape (C) and
+// if the size of the first dimension (dim 0) is the same between Conv
+// input weight and Batchnorm input scale
+static void fuseConvBatchNorm(Block* b, ValueToParamPairMap& valsToParamsMap) {
+  for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end; ++it) {
+    for (auto* child_block : it->blocks()) {
+      fuseConvBatchNorm(child_block, valsToParamsMap);
+    }
+    if (it->kind() == onnx::Conv) {
+      if (it->output()->uses().size() != 1) {
+        continue;
+      }
+      auto bnNode = it->output()->uses()[0].user;
+      if (bnNode->kind() != onnx::BatchNormalization) {
+        continue;
+      }
+      auto origconvNode = *it;
+      auto epsilon = bnNode->f(attr::epsilon);
+      auto w_conv_value = getValues(origconvNode, valsToParamsMap);
+      if (w_conv_value.size() < 1 ||
```

Comment:
The logic here could be stated more clearly.
If inputs are 3, size of w_conv_value should be 2
If inputs are 2, size of w_conv_value should be 1
Continue if none of the above.